File: merge.py
cnt = int(input())
a = []
for i in range(cnt):
    value = int(input())
    a.append(value)

def mergeSort(sorted):

    if len(sorted) <= 1:
        return sorted

    mid = len(sorted) // 2
    left = sorted[:mid]
    right = sorted[mid:]

    left1 = mergeSort(left)
    right1 = mergeSort(right)
    return merge(left1, right1)

def merge(left, right):
    i=0
    j=0
    so = []

while(i < len(left) and j < len(right)):
    if left[i] < right[j]:
        so.append(left[i])
        i += 1
    else:
        so.append(right[j])
        j += 1

while (i < len(left)):
    so.append(left[i])
    i += 1
while (j < len(right)):
    so.append(right[j])
    j += 1

return so

print(mergeSort(a))

File: Merge.cpp
#include <iostream>

using namespace std;

int sorted[8];
int length = 8;

void merge(int *A, int m, int middle, int n)
{
	int i, j, k;
	i = m;
	j = middle + 1;
	k = m;
	
	while(i <= middle && j <= n)
	{
		if(A[i] <= A[j])
		{
			sorted[k] = A[i];
			i++;
		}
		else
		{
			sorted[k] = A[j];
			j++;
		}
		k++;		
	}
	
	if(i > middle)
	{
		for(int t = j; t <= n; t++)
		{
			sorted[k] = A[t];
			k++;
		}
	}
	else
	{
		for(int t = i; t <= middle; t++)
		{
			sorted[k] = A[t];
			k++;
		}
	}
	
	for(int t = m; t <= n; t++)
	{
		A[t] = sorted[t];
	}

}

int mergeSort(int *A, int m, int n)
{ 
	if(m < n)
	{
		int middle = (m + n) / 2;
		
		mergeSort(A, m, middle);
		mergeSort(A, middle+1, n);
		merge(A, m, middle, n);		
	}
	
}

int main()
{
	int A[length]={1,5,2,6,8,9,4,3};
	mergeSort(A, 0, length - 1);
	
	for(int i = 0; i < 8; i++)
	{
		cout << A[i] << " ";
	}
	
}
